### PR TITLE
956 unit tests

### PR DESCRIPTION
**Pull request by abdelr**

### CODE REVIEW

**Review by msevestre — CHANGES_REQUESTED** [review]

File: tests/OSPSuite.Presentation.Tests/Importer/Presenters/ColumnMappingPresenterSpecs.cs [hunk 1]
```diff
@@ -89,41 +97,244 @@ public void identify_basic_format()
...
+      {
+         Assert.AreEqual(_basicFormat.Parameters.OfType<MappingDataFormatParameter>().First(p => p.ColumnName == "Observation").MappedColumn.Unit, _basicFormat.Parameters.OfType<MappingDataFormatParameter>().First(p => p.ColumnName == "Error").MappedColumn.Unit);
+      }
+   }
+
+   public class When_initializing_error_unit_on_initialized_error : ConcernForColumnMappingPresenter
+   {
+      protected override void Because()
+      {
+         A.CallTo(() => _basicFormat.Parameters).Returns(new List<DataFormatParameter>() {
```
[the remark this answers is not in the document]
Comment (abdelr) [reply tests/OSPSuite.Presentation.Tests/Importer/Presenters/ColumnMappingPresenterSpecs.cs]: Done

File: tests/OSPSuite.Presentation.Tests/Importer/Presenters/ColumnMappingPresenterSpecs.cs [hunk 2]
```diff
@@ -89,41 +97,244 @@ public void identify_basic_format()
...
+
+   public class When_getting_available_rows_for : ConcernForColumnMappingPresenter
+   {
+      protected override void Context()
+      {
+         base.Context();
+         A.CallTo(() => _basicFormat.ExcelColumnNames).Returns(new List<string>() { "Time", "Concentration", "Error", "Col1", "Col2" });
+      }
+
+      [TestCase]
```
[the remark this answers is not in the document]
Comment (abdelr) [reply tests/OSPSuite.Presentation.Tests/Importer/Presenters/ColumnMappingPresenterSpecs.cs]: Done

File: src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs [hunk 3]
```diff
@@ -49,11 +48,9 @@ public void ImportDataForConfirmation()
          var sheets = new Cache<string, DataSheet>();
          foreach (var element in _dataSourceFile.DataSheets.KeyValues)
          {
-            if (!Sheets.Keys.Contains(element.Key))
-            {
-               Sheets.Add(element.Key, element.Value);
-               sheets.Add(element.Key, element.Value);
-            }
+            if (Sheets.Keys.Contains(element.Key)) continue;
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs:51]: continue on new line or at least empty line. Too easy to miss.
But I am favouring new line for readability

File: src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs [hunk 4]
```diff
@@ -49,11 +48,9 @@ public void ImportDataForConfirmation()
...
+            Sheets.Add(element.Key, element.Value);
+            sheets.Add(element.Key, element.Value);
          }
 
          if (sheets.Count == 0) return;
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs:56]: same with return when they are in the middle of functions

**Review by msevestre — CHANGES_REQUESTED** [review]
Small style changes. Good to merge otherwise. Please update and I'll merge right away

File: src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs (hunk 4 above)
Comment (abdelr) [reply src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs]: I fixed a couple already. I think it is good to go now. Thanks for the catch.

**Pull request merged**